feature/gribi/otg_tests/ipv4_entry_test/ipv4_entry_test.go

```diff
@@ -288,60 +296,129 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
 }
 
 // configreATE configures port1-3 on the ATE.
-func configureATE(t *testing.T, ate *ondatra.ATEDevice) *ondatra.ATETopology {
-	top := ate.Topology().New()
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) gosnappi.Config {
+	top := ate.OTG().NewConfig(t)
 
 	p1 := ate.Port(t, "port1")
 	p2 := ate.Port(t, "port2")
 	p3 := ate.Port(t, "port3")
 
-	atePort1.AddToATE(top, p1, &dutPort1)
-	atePort2.AddToATE(top, p2, &dutPort2)
-	atePort3.AddToATE(top, p3, &dutPort3)
+	addToOTG(top, p1, &atePort1, &dutPort1)
+	addToOTG(top, p2, &atePort2, &dutPort2)
+	addToOTG(top, p3, &atePort3, &dutPort3)
 
-	top.Push(t).StartProtocols(t)
+	ate.OTG().PushConfig(t, top)
+	ate.OTG().StartProtocols(t)
 
 	return top
 }
 
 // createFlow returns a flow from atePort1 to the dstPfx, expected to arrive on ATE interface dsts.
-func createFlow(name string, ate *ondatra.ATEDevice, ateTop *ondatra.ATETopology, dsts ...*attrs.Attributes) *ondatra.Flow {
-	hdr := ondatra.NewIPv4Header()
-	hdr.WithSrcAddress(dutPort1.IPv4).
-		DstAddressRange().WithMin(dstPfxMin).WithMax(dstPfxMax).WithCount(dstPfxCount)
+func createFlow(t *testing.T, name string, ate *ondatra.ATEDevice, ateTop gosnappi.Config, dsts ...*attrs.Attributes) string {
 
-	endpoints := []ondatra.Endpoint{}
+	// Multiple devices is not supported on the OTG flows
+	modName := strings.Replace(name, " ", "_", -1)
+	rxEndpoints := []string{}
 	for _, dst := range dsts {
-		endpoints = append(endpoints, ateTop.Interfaces()[dst.Name])
+		rxEndpoints = append(rxEndpoints, dst.Name+".IPv4")
 	}
-
-	flow := ate.Traffic().NewFlow(name).
-		WithSrcEndpoints(ateTop.Interfaces()[atePort1.Name]).
-		WithDstEndpoints(endpoints...).
-		WithHeaders(ondatra.NewEthernetHeader(), hdr)
-
-	return flow
+	otg := ate.OTG()
+	flowipv4 := ateTop.Flows().Add().SetName(modName)
+	flowipv4.Metrics().SetEnable(true)
+	e1 := flowipv4.Packet().Add().Ethernet()
+	e1.Src().SetValue(atePort1.MAC)
+	if len(dsts) > 1 {
+		flowipv4.TxRx().Port().SetTxName(atePort1.Name)
+		waitOTGARPEntry(t)
+		dstMac := otg.Telemetry().Interface(atePort1.Name + ".Eth").Ipv4Neighbor(dutPort1.IPv4).LinkLayerAddress().Get(t)
+		e1.Dst().SetChoice("value").SetValue(dstMac)
+	} else {
+		flowipv4.TxRx().Device().SetTxNames([]string{atePort1.Name + ".IPv4"}).SetRxNames(rxEndpoints)
+	}
+	v4 := flowipv4.Packet().Add().Ipv4()
+	v4.Src().SetValue(atePort1.IPv4)
+	v4.Dst().Increment().SetStart(dstPfxMin).SetCount(dstPfxCount)
+	otg.PushConfig(t, ateTop)
+	return modName
 }
 
-func validateTrafficFlows(t *testing.T, ate *ondatra.ATEDevice, good []*ondatra.Flow, bad []*ondatra.Flow) {
+func validateTrafficFlows(t *testing.T, ate *ondatra.ATEDevice, good []string, bad []string) {
```

Comment:
i don't quite follow the logic here / what is a good flow / bad flow?

Reply:
I tried to follow the logic used in the ATE test with minimal changes. The good flow should run with 0 packet loss where the bad flow is expected to have 0 rx packets